### DISCUSSION TITLE
Is there a `belongsToMany` decorator in Adonis V5?

### DISCUSSION THREAD

**Discussion by mlbonniec** [adonisjs/core#3164] [Help] [answered]
Hi ^^

I'm looking for a `belongsToMany` decorator in Adonis V5.
Here's my database structure. I would like to get all participants and messages from a conversations, but it seems impossible.

<div align="center">
<img width="722" alt="Capture d’écran 2021-09-13 à 11 15 26" src="https://user-images.githubusercontent.com/29955402/133057807-7fb021e0-ecf6-4ec7-ae1e-36dcd531652a.png">
</div>

Actually, my `Conversations` model is this. The problem is that I get only one participant, not all.
```ts
class Conversations extends BaseModel {
  @column({ isPrimary: true })
  public id: BigInt;

  @belongsTo(() => ConversationsParticipants, { localKey: 'conversationId', foreignKey: 'id' })
  public participants: BelongsTo<typeof ConversationsParticipants>;

  // Same for messages in the future

  @column.dateTime({ autoCreate: true })
  public dateOfCreation: DateTime;
}
```

How can I do?
Thanks

**Comment by DraftProducts** [comment, marked as answer]
Hope you found your answer yourself, otherwise, the decorator you can use in your case is [`manyToMany`](https://docs.adonisjs.com/guides/models/relationships#many-to-many) to get directly your profiles through your pivot table.
```ts
@manyToMany(() => Profiles, {
  pivotTable: 'conversations_participants',
  localKey: 'id',
  pivotForeignKey: 'profile_id',
  pivotRelatedForeignKey: 'conversation_id',
  relatedKey: 'id',
})
public participants: ManyToMany<typeof Profiles>;
```
*If you follow the naming convention of models, which is to not pluralize their name:*
```ts
@manyToMany(() => Profile, { ... })
public participants: ManyToMany<typeof Profile>;
```

**Reply by mlbonniec** [reply]
Yep, that's what I used.
Thanks for your help anyway 😉

I'll change this, following the convention :)
